realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.kt

```diff
@@ -679,12 +679,13 @@ class RealmProxyClassGenerator(private val processingEnvironment: ProcessingEnvi
                 val persistedFields = metadata.fields.size
                 val computedFields = metadata.backlinkFields.size
                 val embeddedClass = if (metadata.embedded) "true" else "false"
-
-                emitStatement("OsObjectSchemaInfo.Builder builder = new OsObjectSchemaInfo.Builder(\"%s\", %s, %s, %s)", internalClassName, embeddedClass, persistedFields, computedFields)
+                val publicClassName = if (simpleJavaClassName.name != internalClassName) simpleJavaClassName.name else ""
+                emitStatement("OsObjectSchemaInfo.Builder builder = new OsObjectSchemaInfo.Builder(\"%s\", \"%s\", %s, %s, %s)", publicClassName, internalClassName, embeddedClass, persistedFields, computedFields)
```

Comment:
Shouldn't we use string templating?
```suggestion
                emitStatement("OsObjectSchemaInfo.Builder builder = new OsObjectSchemaInfo.Builder(\"${publicClassName}\", \"${internalClassName}\", ${embeddedClass}, ${persistedFields}, ${computedFields})")
```